### PR TITLE
[WIP] Fix clash card, add condition logic example

### LINKED ISSUE #93: Clash card doesn't do what it says
The `Clash` card says it can only be played if "every card in your hand is an attack". But it's not actually implemented.

https://github.com/oskarrough/slaytheweb/blob/9daa6ced995d4b087a6a1d2d16e44ded522c3c0c/public/game/cards.js#L35-L42

We can either

a. remove/change the card
b. implement it

If we implement, how? Maybe we need a `utils.canPlayCard(state, card)` method. But then all cards would have to share logic, which isn't the case. Cards need to be able to define their own logic. `card.canPlay()`?

### PR DESCRIPTION
**Pull request by SamuelRott**
This could be a starting solution for https://github.com/oskarrough/slaytheweb/issues/93.

Fixes #93

### CODE REVIEW

File: public/ui/cards.js [hunk 1]
```diff
@@ -5,14 +5,37 @@ export default class Cards extends Component {
...
+				${props.cards.map(card => Card(card, props.energy, props.cards))}
 			</div>
 		`
 	}
 }
 
-export function Card(card, currentEnergy) {
-	const isDisabled = currentEnergy < card.energy
+function isCardDisabled(card, currentEnergy, cards) {
+	let isDisabled;
```
Comment (oskarrough) [inline public/ui/cards.js:15]:
```suggestion
	let isDisabled = false
```

to make it a tiny bit clearer

File: public/ui/cards.js [hunk 2]
```diff
@@ -5,14 +5,37 @@ export default class Cards extends Component {
...
+	if(currentEnergy < card.energy) {
+		isDisabled = true
+	} 
+	// Example of how we could handle cards conditions
+	// Should probably be a isolated function
```
Comment (oskarrough) [inline public/ui/cards.js:20]: think it's cool like this for now. we can always complicate it later

**Review by oskarrough — APPROVED** [review]
Very nice, thanks again. I think we should just go ahead with this.

Can we think of other examples for conditions to see if it is flexible enough?

**Comment by oskarrough** [comment]
Here's another idea. All cards could _optionally_ have a `canUse(state)` method that returns true/false. This way…

1.  the logic stay with the card
2. the method would be passed the current game `state`
3. you can do whatever you want because code

If more cards follow the same logic we could refactor into some external functions but not even sure that's needed. 

One downside I'm not sure about is that currently cards are complete function-less. It's just an object with properties. Does it  matter?

**Comment by SamuelRott** [comment]
_Can we think of other examples for conditions to see if it is flexible enough?_ => Yes, this would be my first objective, try to have a few cases to see where we quickly hit the wall. This canUse/Conditions logic could become a very complex part of the game.

_Here's another idea / canUse_ => I would a full demonstration or more explanation to fully understand what you mean :).

_It's just an object with properties. Does it matter?_ => Idk and I dont think so. I think this could be answered when we face more complexity.

If the fix for clash is not urgent I would wait one or two days to merge, the time to come up with more scenarios.

**Comment by oskarrough** [comment]
I merge now, then we can discuss further in peace :)

**Pull request merged**